pkg/chunk/aws/s3_storage_client.go

```diff
@@ -38,8 +43,24 @@ func init() {
 // S3Config specifies config for storing chunks on AWS S3.
 type S3Config struct {
 	S3               flagext.URLValue
-	BucketNames      string
 	S3ForcePathStyle bool
+
+	BucketNames     string
+	Endpoint        string             `yaml:"endpoint"`
+	Region          string             `yaml:"region"`
+	AccessKeyID     string             `yaml:"access_key_id"`
+	SecretAccessKey string             `yaml:"secret_access_key"`
+	Insecure        bool               `yaml:"insecure"`
+	SSEEncryption   bool               `yaml:"sse_encryption"`
+	PutUserMetadata map[string]*string `yaml:"put_user_metadata"`
+	HTTPConfig      HTTPConfig         `yaml:"http_config"`
```

Comment:
Is it worth a nested config struct instead of embedding the 3 config options here?

Reply:
I don't have strong feelings here.  This was admittedly modeled after the thanos s3 client (https://github.com/thanos-io/thanos/tree/master/pkg/objstore/s3).  

If you'd like me to embed them in the S3Config struct I'd be glad to.